### PR TITLE
Add what's new section for 4.0

### LINKED ISSUE #180: 4.0: Add what's new in 4.0 section
There should be a top-level section that summarizes the major changes from a user perspective, with links to pages that contain detail.

### PR DESCRIPTION
**Pull request by dtrudg**
Closes #180

### CODE REVIEW

File: new.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+OCI-mode
+********
+
+Singularity 4 introduces :ref:`OCI-mode <oci_mode>` as a fully supported
+feature. It is enabled by using the ``--oci`` flag with the run / shell / exec /
+pull commands, or when ``oci mode = yes`` is set in ``singularity.conf``.
+
+In OCI-mode:
+
+- Container images from OCI sources will be pull-ed to an :ref:`OCI-SIF
```
Comment (preminger) [inline new.rst]: "pull-ed" looks sort of weird here... Maybe change to "`pull`-ed" or "pulled"?

File: new.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  <oci_sif>` file. An OCI-SIF file encapsulates the OCI image configuration and
+  squashed filesystem using an OCI, rather than Singularity specific, structure.
+- The run / shell / exec commands use a low-level OCI runtime (crun/runc) for
+  container execution.
+- Default operation is :ref:`compatible with other OCI tools <oci_compat>`,
+  similar to ``--compat`` in Singularity's non-OCI native mode.
+- OCI-mode supports running existing Singularity non-OCI-SIF images, and can be
```
Comment (preminger) [inline new.rst]: "non-OCI-SIF" is ambiguous, I think, between "[non-OCI]-SIF" (=SIFs that are non-OCI) and "non-[OCI-SIF]" (=images that are not OCI-SIF) - the latter being what (I think) was intended...? In any event, I suggest rephrasing in a way that's not ambiguous, perhaps using the parenthesized text above.

File: new.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+***********
+
+- The commands related to OCI/Docker registries that were under ``singularity
+  remote`` have been moved to their own, dedicated :ref:`registry command
+  <registry>`.
+- The keyserver-related commands that were under ``singularity remote`` have
+  been moved to their own, dedicated :ref:`keyserver command <key_commands>`.
+- Adding a new remote endpoint using the ``singularity remote add`` command will
+  now set the new endpoint as default. This behavior can be suppressed by
+  supplying the ``--no-default`` (or ``-n``) flag to remote add.
```
Comment (preminger) [inline new.rst]:
```suggestion
  supplying the ``--no-default`` (or ``-n``) flag to ``remote add``.
```

File: new.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  file unchanged (with its homedir field pointing to ``/root``, regardless of the
+  value passed to ``--home)``. Now, both the value of HOME and the
+  contents of ``/etc/passwd`` in the container will reflect the value passed to
+  ``--home``.
+- Bind mounts are now performed in the order of their occurrence on the command
+  line, or within the value of the ``SINGULARITY_BIND`` environment variable.
+  (Previously, image-mounts were always performed first, regardless of order.)
+- The current working directory is created in the container when it doesn't
+  exist, so that it can be entered. You must now specify ``--no-mount home,cwd``
+  instead of just ``--no-mount`` home to avoid mounting from ``$HOME`` if you
```
Comment (preminger) [inline new.rst]:
```suggestion
  instead of just ``--no-mount home`` to avoid mounting from ``$HOME`` if you
```

File: new.rst [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+- Added ``--private`` as a synonym for ``--secret`` in ``key list``, ``key
+  export``, and ``key remove`` subcommands.
+- The ``instance start`` command now accepts an optional ``--app <name>``
+  argument which invokes the start script within the ``%appstart <name>``
+  section in the definition file. The ``instance stop`` command still only
+  requires the instance name.
+- A new ``--no-pid`` flag for ``singularity run/shell/exec`` disables the PID
+  namespace inferred by ``--containall`` and ``--compat``.
+- :ref:`Caching of OCI images <sec:cache>` is now architecture aware. This fixes
+  behavior where a user's home directory is shared between systems of different
```
Comment (preminger) [inline new.rst]:
```suggestion
  behavior in cases where a user's home directory is shared between systems of different
```

File: new.rst [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  architectures. If you do not use older versions of {Singularity} on a system,
```
Comment (preminger) [inline new.rst]:
```suggestion
  architectures. If you do not use older versions of {Singularity} on the same system,
```

**Review by preminger — APPROVED** [review]
LGTM - some suggestions below.

File: new.rst (hunk 1 above)
Comment (dtrudg) [reply new.rst]: Ahh yep, it was meant to be `pull`-ed. Forgot to add back the formatting after copy and pasting.

**Pull request merged**